fabric-common/lib/BlockDecoder.js

```diff
@@ -1140,496 +1242,395 @@ function decodeFabricOUIdentifier(proto_organizational_unit_identitfiers) {
 	}
 
 	return kv_rw_set;
 }
 
-function decodeKVRead(proto_kv_read) {
-	const kv_read = {};
-	kv_read.key = proto_kv_read.getKey();
-	const proto_version = proto_kv_read.getVersion();
-	if (proto_version) {
-		kv_read.version = {};
-		kv_read.version.block_num = proto_version.getBlockNum().toString();
-		kv_read.version.tx_num = proto_version.getTxNum().toString();
-	} else {
-		kv_read.version = null;
+function decodeVersion(versionProto) {
+	if (versionProto) {
+		const version = {};
+		version.block_num = versionProto.block_num;
+		version.tx_num = versionProto.tx_num;
+		return version;
 	}
 
+	return;
+}
+
+function decodeKVRead(kVReadProto) {
+	const kv_read = {};
+	kv_read.key = kVReadProto.key;
+	kv_read.version = decodeVersion(kVReadProto.version);
+
 	return kv_read;
 }
 
-function decodeRangeQueryInfo(proto_range_query_info) {
+function decodeRangeQueryInfo(rangeQueryInfoProto) {
 	const range_query_info = {};
-	range_query_info.start_key = proto_range_query_info.getStartKey();
-	range_query_info.end_key = proto_range_query_info.getEndKey();
-	range_query_info.itr_exhausted = proto_range_query_info.getItrExhausted();
+	range_query_info.start_key = rangeQueryInfoProto.start_key;
+	range_query_info.end_key = rangeQueryInfoProto.end_key;
+	range_query_info.itr_exhausted = rangeQueryInfoProto.itr_exhausted;
 
-	// reads_info is one of QueryReads
-	const proto_raw_reads = proto_range_query_info.getRawReads();
-	if (proto_raw_reads) {
+	// reads_info is one of QueryReads or QueryReadsMerkleSummary
+	if (rangeQueryInfoProto.raw_reads) {
 		range_query_info.raw_reads = {};
 		range_query_info.raw_reads.kv_reads = [];
-		for (const i in proto_raw_reads.kv_reads) {
-			const kv_read = decodeKVRead(proto_raw_reads.kv_reads[i]);
-			range_query_info.raw_reads.kv_reads.push(kv_read);
+		for (const kVReadProto of rangeQueryInfoProto.raw_reads) {
+			range_query_info.raw_reads.kv_reads.push(decodeKVRead(kVReadProto));
 		}
-	}
-	// or QueryReadsMerkleSummary
-	const proto_reads_merkle_hashes = proto_range_query_info.getReadsMerkleHashes();
-	if (proto_reads_merkle_hashes) {
+	} else if (rangeQueryInfoProto.reads_merkle_hashes) {
 		range_query_info.reads_merkle_hashes = {};
-		range_query_info.reads_merkle_hashes.max_degree = proto_reads_merkle_hashes.getMaxDegree();
-		range_query_info.reads_merkle_hashes.max_level = proto_reads_merkle_hashes.getMaxLevel();
-		range_query_info.reads_merkle_hashes.max_level_hashes = proto_reads_merkle_hashes.getMaxLevelHashes();
+		range_query_info.reads_merkle_hashes.max_degree = rangeQueryInfoProto.reads_merkle_hashes.max_degree;
+		range_query_info.reads_merkle_hashes.max_level = rangeQueryInfoProto.reads_merkle_hashes.max_level;
+		range_query_info.reads_merkle_hashes.max_level_hashes = rangeQueryInfoProto.reads_merkle_hashes.max_level_hashes;
 	}
 
 	return range_query_info;
 }
 
-function decodeKVWrite(proto_kv_write) {
+function decodeKVWrite(kVWriteProto) {
```

Comment:
same as above

Reply:
Yes we could do that, we should do that with an overall clean up of this class